### PR TITLE
fix filter removal for http filter

### PR DESCRIPTION
**Pull request by ramaraochavali**
Found this bug in this PR https://github.com/istio/istio/pull/36404. Http filter removal does not work correctly.

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch.go [hunk 1]
```diff
@@ -512,6 +520,7 @@ func patchHTTPFilters(patchContext networking.EnvoyFilter_PatchContext,
 			if removedFilters.Contains(filter.Name) {
 				continue
 			}
+			tempArray = append(tempArray, filter)
```
Comment (ramaraochavali) [inline pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch.go:523]: This is the actual fix. Not adding to tempArray. Rest of the changes are format changes.

File: pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch.go [hunk 2]
```diff
@@ -526,7 +535,8 @@ func patchHTTPFilters(patchContext networking.EnvoyFilter_PatchContext,
 func patchHTTPFilter(patchContext networking.EnvoyFilter_PatchContext,
 	patches map[networking.EnvoyFilter_ApplyTo][]*model.EnvoyFilterConfigPatchWrapper,
 	listener *xdslistener.Listener, fc *xdslistener.FilterChain, filter *xdslistener.Filter,
-	httpFilter *hcm.HttpFilter) bool {
+	httpFilter *hcm.HttpFilter,
+) bool {
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch.go:539]: can you use the older `gofumpt` to avoid noise in PRs?

**Review by howardjohn — COMMENTED** [review]
do we have any tests for this?

File: pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch.go:523]: Can you add a test? Lack of ut cause this happen.

**Comment by ramaraochavali** [comment]
network filter removal has test but not http filter removal. Added a test for it. PTAL.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**